src/main/java/frc/robot/hardware/mechanisms/MechanismSimulation.java

```diff
@@ -2,29 +2,21 @@
 
 import edu.wpi.first.math.geometry.Rotation2d;
 
-public abstract class MechanismSimulation {
+public interface MechanismSimulation {
 
-	private final double gearRatio;
+	double getGearRatio();
 
-	MechanismSimulation(double gearRatio) {
-		this.gearRatio = gearRatio;
-	}
+	Rotation2d getRotorPosition();
 
-	public Rotation2d getRotorPosition() {
-		return getSystemPosition().times(gearRatio);
-	}
+	Rotation2d getRotorVelocityAnglesPerSecond();
```

Comment:
angles per seconds? Maybe rotation2d per seconds. Angles are not units, but seconds are. It physically doesn't work. 

Also, angles should be in singular and not plural. @YHerm